flask_resty/decorators.py

```diff
@@ -7,6 +7,20 @@
 
 
 def get_item_or_404(func=None, **decorator_kwargs):
+    """
+    Similar to Django's :py:func:`django.shortcuts.get_object_or_404` but
+    as a decorator. Decorating :py:meth:`ApiView.retrieve`,
+    :py:meth:`ApiView.update` or :py:meth:`ApiView.destroy` replaces the `id`
+    parameter of the decorated method with the `item` corresponding to the
+    provided `id`.
+
+    Behaves like a decorator factory if `func` is omitted.
+
+    :param func: The function to decorate
+    :type func: function or None
+    :return: The decorated function or a decorator factory
+    :rtype: function
+    """
     # Allow using this as either a decorator or a decorator factory.
     if func is None:
```

Comment:
@taion, just noticing this now, this is a rather bizarre behavior for a function to return such different outputs in case a param is passed or not. I'd say that `functools.partial` should be done in userland. This is pretty pedantic I guess, but I just wanted to know your opinion about it

Reply:
i see this probably a slight majority of the time for decorators with optional arguments

e.g. we quite often use `@pytest.fixture`, which works... don't need to do the full `@pytest.fixture()`. 

click always requires the explicit function call though. idk

Reply:
oh yeah, in the context of decorators, it makes sense I guess